src/modules/75-cf/components/AddKeyDialog/__tests__/AddKeyDialog.test.tsx

```diff
@@ -44,13 +69,86 @@ describe('Test AddKeyDialog', () => {
             projectIdentifier: 'dummy'
           }}
           onCreate={onCreate}
+          {...props}
         />
       </TestWrapper>
     )
-    const { container, rerender } = render(Component)
+  }
+
+  const openDialog = async (): Promise<void> => {
+    const addKeyBtn = screen.getByRole('button', { name: 'cf.environments.apiKeys.addKey' })
+    expect(addKeyBtn).toBeInTheDocument()
+    userEvent.click(addKeyBtn)
+    await waitFor(() => expect(screen.getByText('cf.environments.apiKeys.addKeyTitle')).toBeInTheDocument())
+  }
+
+  test('AddKeyDialog should be rendered properly', async () => {
+    const { container } = renderComponent()
+
+    userEvent.click(container.querySelector('button') as HTMLButtonElement)
+
+    await waitFor(() => expect(container).toMatchSnapshot())
+  })
+
+  test('it should render duplicate SDK key name error message when user enters an existing SDK key name', async () => {
+    renderComponent({ apiKeys: apiKeysMock })
+
+    await openDialog()
+
+    userEvent.type(document.querySelector('input[name=name]') as HTMLInputElement, 'dummy api key name')
+    userEvent.click(screen.getByRole('button', { name: 'createSecretYAML.create' }))
+
+    await waitFor(() => {
+      expect(screen.getByText('cf.environments.apiKeys.duplicateKey')).toBeInTheDocument()
+    })
+  })
+
+  test('it should render missing SDK key name error message when user attempts to submit form without entering a name', async () => {
+    renderComponent()
+
+    await openDialog()
+
+    userEvent.click(screen.getByText('createSecretYAML.create'))
+
+    await waitFor(() => {
+      expect(screen.getByText('cf.environments.apiKeys.emptyName')).toBeInTheDocument()
+    })
+  })
+  test('it should send correct values to useAddAPIKey and display success message', async () => {
```

Comment:
Could do with a line break to separate these tests.
```suggestion
  
  test('it should send correct values to useAddAPIKey and display success message', async () => {
```

Reply:
done